scalalib/src/mill/scalalib/cosmo/CosmoModule.scala

```diff
@@ -0,0 +1,155 @@
+package mill.scalalib.cosmo
+
+import mill.{T, Task}
+import mill.api.{PathRef, Result}
+import mill.scalalib._
+
+/**
+ * Provides a [[CosmoModule.cosmoAssembly task]] to build a cross-platfrom executable assembly jar
+ * by prepending an [[https://justine.lol/ape.html Actually Portable Executable (APE)]]
+ * launcher binary compiled with the [[https://justine.lol/cosmopolitan/index.html Cosmopolitan Libc]]
+ */
+trait CosmoModule extends mill.Module with AssemblyModule {
+  def finalMainClass: T[String]
+
+  def forkArgs(argv0: String): Task[Seq[String]] = Task.Anon { Seq[String]() }
+
+  def cosmoccVersion: T[String] = Task { "" }
+
+  def cosmocc: T[PathRef] = Task(persistent = true) {
+    val version = if (cosmoccVersion().isEmpty) "" else s"-${cosmoccVersion()}"
+    val versionedCosmocc = s"cosmocc${version}"
+    val zip = Task.dest / s"${versionedCosmocc}.zip"
+    val dest = Task.dest / versionedCosmocc / "bin" / "cosmocc"
+
+    if (!os.exists(dest)) {
+      os.write.over(
+        zip,
+        requests.get.stream(s"https://cosmo.zip/pub/cosmocc/${versionedCosmocc}.zip")
+      )
+      os.remove.all(Task.dest / versionedCosmocc)
+      os.call(("unzip", zip, "-d", Task.dest / versionedCosmocc))
```

Comment:
Can you update the issue upstream in OS-Lib with the details of what doesn't work, and link the issue in a comment here?

Reply:
Use `os.unzip` now. Turns out I need to `+x` files inside `libexec` too, not just `bin`. Also commented on the issue with more details.